.github/workflows/go-tests.yml

```diff
@@ -0,0 +1,43 @@
+name: go-tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  merge_group:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  FOUNDRY_PROFILE: ci
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
```

Comment:
Maybe we should just merge the contracts-tests and go-tests into one `main.yml` file to make it easier to maintain. If we wanted to keep them separate we can add a setup action like we do in gateway and ecosystem that pulls in github and foundry

Reply:
done